### PR TITLE
Makes ability buttons and grabs unscannable by the chameleon projector

### LINKED ISSUE #9184: You can scan ability buttons with the chameleon projector
### Labels

[minor]

### Description

Scanning an ability button allows you to chameleon project as said ability button

### Steps to reproduce

1. Have something like PDA light or superfart
2. Click the button while holding a chameleon projector.
3. Be superfart.

### Expected Behavior

To use the ability instead of scanning it.

### Additional Information & Screenshots

image

### LINKED ISSUE #9186: You can scan grabs with the chameleon projector
### Labels

[minor]

### Description

You can scan a grab and become a grab

### Steps to reproduce

1. Grab somebody with an empty hand
2. Click on the grab with a chameleon projector
3. Become a grab

### Expected Behavior

To not be able to scan a grab.

### Additional Information & Screenshots

image

### LINKED ISSUE #9381: Chameleon Projector can scan UI buttons
### Labels

[BUG]

### Description

The chameleon projector can scan UI buttons.

### Steps to reproduce

1. Be a traitor
2. buy a chameleon projector
3. click on your Internals button or PDA light with it in hand
4. turn on the projector
5. be UI

### Expected Behavior

To not make yourself become UI

### Additional Information & Screenshots

dreamseeker_8jMVJMd57j

### PR DESCRIPTION
**Pull request by DimWhat**
[minor]
## About the PR 
Fixes #9184, fixes #9186, and fixes #9381. Should let players stop scanning some ui elements and being them.

## Why's this needed? 
While amusing to some, seeing UI stuff outside of the UI is really jarring.

### CODE REVIEW

**Comment by ZeWaka** [comment]
Label your PRs

**Comment by ZeWaka** [comment]
Label your PRs

**Comment by DimWhat** [comment]
> Label your PRs

Apologies, i assumed it worked when i first put the tag up top.

File: code/obj/item/device/chameleon.dm [hunk 1]
```diff
@@ -122,6 +122,8 @@
 			if (user && ismob(user))
 				user.show_text("You are too far away to do that.", "red")
 			return
+		if (istype(target, /obj/ability_button) || istype(target, /obj/item/grab)) //just don't scan ability buttons or grabs
```
Comment (ZeWaka) [inline code/obj/item/device/chameleon.dm:125]: ...is there not a var or helper proc to check instead?

Comment (DimWhat) [reply code/obj/item/device/chameleon.dm:125]: There is one for hud stuff! Just not sure if there is one for grabs.

File: code/obj/item/device/chameleon.dm [hunk 2]
```diff
@@ -122,6 +122,8 @@
 			if (user && ismob(user))
 				user.show_text("You are too far away to do that.", "red")
 			return
+		if (plane == PLANE_HUD || istype(target, /obj/item/grab)) //just don't scan hud stuff or grabs
```
Comment (ZeWaka) [inline code/obj/item/device/chameleon.dm:125]:
```suggestion
		if (plane == PLANE_HUD || isgrab(target)) //just don't scan hud stuff or grabs
```

File: code/obj/item/device/chameleon.dm [hunk 3]
```diff
@@ -236,6 +238,8 @@
 			if (user && ismob(user))
 				user.show_text("You are too far away to do that.", "red")
 			return
+		if (plane == PLANE_HUD  || istype(target, /obj/item/grab)) //just don't scan hud stuff and grabs
```
Comment (ZeWaka) [inline code/obj/item/device/chameleon.dm:241]:
```suggestion
		if (plane == PLANE_HUD  || isgrab(target)) //just don't scan hud stuff and grabs
```

**Review by ZeWaka — APPROVED** [review]

**Pull request merged**